crates/bevy_render/src/mesh/mesh/conversions.rs

```diff
@@ -0,0 +1,362 @@
+//! This implementations allows you
+//! convert std::vec::Vec<T> to VertexAttributeValues::T and back.
+//!
+//! # Examples
+//!
+//! ```rust
+//! use std::convert::TryInto;
+//! use bevy_render::mesh::VertexAttributeValues;
+//!
+//! // creating vector of values
+//! let before = vec![[0_u32; 4]; 10];
+//! let values = VertexAttributeValues::from(before.clone());
+//! let after: Vec<[u32; 4]> = values.try_into().unwrap();
+//!
+//! assert_eq!(before, after);
+//! ```
+
+use crate::mesh::VertexAttributeValues;
+use std::convert::TryInto;
+
+const CANT_CONVERT: &str = "can't convert to ";
+
+impl From<Vec<f32>> for VertexAttributeValues {
+    fn from(vec: Vec<f32>) -> Self {
+        VertexAttributeValues::Float(vec)
+    }
+}
+
+impl From<Vec<i32>> for VertexAttributeValues {
+    fn from(vec: Vec<i32>) -> Self {
+        VertexAttributeValues::Int(vec)
+    }
+}
+
+impl From<Vec<u32>> for VertexAttributeValues {
+    fn from(vec: Vec<u32>) -> Self {
+        VertexAttributeValues::Uint(vec)
+    }
+}
+
+impl From<Vec<[f32; 2]>> for VertexAttributeValues {
+    fn from(vec: Vec<[f32; 2]>) -> Self {
+        VertexAttributeValues::Float2(vec)
+    }
+}
+
+impl From<Vec<[i32; 2]>> for VertexAttributeValues {
+    fn from(vec: Vec<[i32; 2]>) -> Self {
+        VertexAttributeValues::Int2(vec)
+    }
+}
+
+impl From<Vec<[u32; 2]>> for VertexAttributeValues {
+    fn from(vec: Vec<[u32; 2]>) -> Self {
+        VertexAttributeValues::Uint2(vec)
+    }
+}
+
+impl From<Vec<[f32; 3]>> for VertexAttributeValues {
+    fn from(vec: Vec<[f32; 3]>) -> Self {
+        VertexAttributeValues::Float3(vec)
+    }
+}
+
+impl From<Vec<[i32; 3]>> for VertexAttributeValues {
+    fn from(vec: Vec<[i32; 3]>) -> Self {
+        VertexAttributeValues::Int3(vec)
+    }
+}
+
+impl From<Vec<[u32; 3]>> for VertexAttributeValues {
+    fn from(vec: Vec<[u32; 3]>) -> Self {
+        VertexAttributeValues::Uint3(vec)
+    }
+}
+
+impl From<Vec<[f32; 4]>> for VertexAttributeValues {
+    fn from(vec: Vec<[f32; 4]>) -> Self {
+        VertexAttributeValues::Float4(vec)
+    }
+}
+
+impl From<Vec<[i32; 4]>> for VertexAttributeValues {
+    fn from(vec: Vec<[i32; 4]>) -> Self {
+        VertexAttributeValues::Int4(vec)
+    }
+}
+
+impl From<Vec<[u32; 4]>> for VertexAttributeValues {
+    fn from(vec: Vec<[u32; 4]>) -> Self {
+        VertexAttributeValues::Uint4(vec)
+    }
+}
+
+impl From<Vec<[u8; 4]>> for VertexAttributeValues {
+    fn from(vec: Vec<[u8; 4]>) -> Self {
+        VertexAttributeValues::Uchar4Norm(vec)
+    }
+}
+
+impl TryInto<Vec<[u8; 4]>> for VertexAttributeValues {
+    type Error = String;
```

Comment:
I think we should have a
```rust
enum FromVertexAttributeError {
  format: &'static str, // e.g. Uchar4Norm
  into: &'static str, // e.g. Vec<[u8; 4]>
}
```
with the `std::error::Error`, `Display` and `Debug` implementations instead of a String error type.

The `format` field would allow for nicer error messages like ``can't convert `UChar4Norm` vertex attribute to a `Vec<[u8; 4]>` ``